Example/AblyChatExample/ContentView.swift

```diff
@@ -1,21 +1,143 @@
-            Text("Hello, world!")
+            Text(title)
+                .font(.headline)
+                .padding(5)
+            Text(reactions)
+                .frame(maxWidth: .infinity, alignment: .leading)
+                .padding(5)
+            List(messages, id: \.timeserial) { message in
+                MessageBasicView(message: message)
+                    .flip()
+            }
+            .flip()
+            .listStyle(PlainListStyle())
+            HStack {
+                TextField("Type a message...", text: $newMessage)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+                
+                Button(action: {
+                    Task {
+                        try await sendMessage()
+                    }
+                }) {
+#if os(iOS)
+                    Text("Send")
+                        .foregroundColor(.white)
+                        .padding(.vertical, 6)
+                        .padding(.horizontal, 12)
+                        .background(Color.blue)
+                        .cornerRadius(15)
+#elseif os(macOS)
+                    Text("Send")
+#endif
+                }
+                Button(action: {
+                    Task {
+                        try await sendReaction(type: "like")
+                    }
+                }) {
+                    Text("👍")
+                }
+            }
+            .padding(.bottom, 10)
+            .padding(.horizontal, 12)
+            .task {
+                await setDefaultTitle()
+            }
+            .task {
+                for await message in await room().messages.subscribe(bufferingPolicy: .unbounded) {
+                    withAnimation {
+                        messages.insert(message, at: 0)
+                    }
+                }
+            }
+            .task {
+                for await reaction in await room().reactions.subscribe(bufferingPolicy: .unbounded) {
+                    withAnimation {
+                        reactions.append(reaction.type == "like" ? "👍" : "🤷")
+                    }
+                }
+            }
+            .task {
+                for await typing in await room().typing.subscribe(bufferingPolicy: .unbounded) {
+                    withAnimation {
+                        title = "Typing: \(typing.currentlyTyping.joined(separator: ", "))"
+                        Task {
+                            try? await Task.sleep(nanoseconds: 1 * 1_000_000_000)
+                            await setDefaultTitle()
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    func setDefaultTitle() async {
+        title = await "Room: \(room().roomID)"
+    }
+    
+    func sendMessage() async throws {
+        guard !newMessage.isEmpty else { return }
+        let message = try await room().messages.send(params: .init(text: newMessage))
+        withAnimation {
+            messages.insert(message, at: 0)
```

Comment:
Isn't this message going to end up being displayed twice (one here and another time when it's received from the chat client)? I saw that you mentioned elsewhere that it's customary to display the message as soon as the user sends it, but I think that for now it's better to have a delay than it is for the message to appear twice. Same question for reactions.

Reply:
https://github.com/ably-labs/ably-chat-swift/pull/34/commits/b4d7de03074f3f737d8f578d85b34e3d4aacb268